src/utils.ts

```diff
@@ -24,7 +24,9 @@ export class SingleInstanceTimeout {
   public timeout(fn: (...args: any[]) => void, delay: number, ...args: any[]): void {
     if (!this.timerLock) {
       this.timerLock = true;
-      this.timerId = setTimeout(
+      // temporary fix until we use platform specific extensions and can remove this class
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      this.timerId = setTimeout<any[]>(
```

Comment:
This class is removed when we bundle terraform-ls, so I don't see the need for fixing this now. I think the work to change the tsconfig is a separate set of work than what this PR aims to do.

Reply:
I'm OK with keeping the temporary fix in here.

But let me explain why I think it's essential to update the `tsconfig.json` with the Node and TypeScript update. `es2020` is the [recommended target](https://github.com/tsconfig/bases/blob/main/bases/node14.json) since Node 14. With the [new TypeScript 4.5 feature](https://devblogs.microsoft.com/typescript/announcing-typescript-4-5-beta/#supporting-lib-from-node_modules) I think it's important to have a value defined for `lib`, too.
Additionally, the template for new extensions is setting these [values](https://github.com/microsoft/vscode-generator-code/blob/main/generators/app/templates/ext-command-ts/tsconfig.json), too, so we shouldn't get a problem doing it the same.

What do you think?

Reply:
I'm convinced!